### ISSUE TITLE
documentTypeList's that show custom child views render incorrectly in nested views

### ISSUE THREAD

**Issue by ndimatteo** [sanity-io/sanity#2196]
**Describe the bug**

In `deskStructure.js`, if you nest a `documentTypeList()` with `child()` views and go to create a **new document** you lose your open panes and the new document opens at the root of the desk and none of the "views" are available.

**To Reproduce**

If you nest a list like this:
```javascript
S.listItem()
  .title('Collections')
  .child(S.documentTypeList('collection').title('Collections'))
```
it works fine in nested views, but if you nest like this:
```javascript
S.listItem()
  .title('Collections')
  .schemaType('collection')
  .child(
    S.documentTypeList('collection')
      .title('Collections')
      .child(documentId =>
        S.document()
          .documentId(documentId)
          .schemaType('collection')
          .views([
            S.view.form().icon(EditIcon),
            S.view
              .component(SeoPreview)
              .options({ previewURL })
              .icon(EyeIcon)
              .title('SEO Preview')
          ])
      )
  ),
```
the experience breaks. New documents are missing the specified views and any panes that were previously open are collapsed, removing the editor from the context in which it was opened from. So it goes from this:
![image](https://user-images.githubusercontent.com/737188/102681589-9a044800-4190-11eb-86ca-f3ece7fe53d5.png)
To this:
![image](https://user-images.githubusercontent.com/737188/102681521-06327c00-4190-11eb-8d81-b7400f529914.png)

**Expected behavior**

Creating a new document from a nested `documentList` should open and render the same, whether you render it the simple way, or with custom views.

**Which versions of Sanity are you using?**
```
@sanity/cli             2.1.0 (up to date)
@sanity/base            2.1.3 (up to date)
@sanity/cli             2.1.0 (up to date)
@sanity/components      2.1.0 (up to date)
@sanity/core            2.1.2 (up to date)
@sanity/dashboard       2.1.0 (up to date)
@sanity/default-layout  2.1.3 (up to date)
@sanity/default-login   2.1.0 (up to date)
@sanity/desk-tool       2.1.3 (up to date)
@sanity/vision          2.1.0 (up to date)
```

**What operating system are you using?**

MacOS Catalina 10.15.7 (19H15)

**Which versions of Node.js / npm are you running?**
```
6.14.8
v12.13.1
```

**Additional context**

I think something is going on with the URL it opens when you create a new document this way. It fails to retain the current active pane URL path you're on, and instead opens the new document at your root studio URL (check the screenshot above to see the URL

**Comment by ndimatteo** [comment]
Any chance you've had a chance to review this one @kmelve ?

Still seeing this persist which is very jarring for content editors 🙃

**Comment by kmelve** [comment]
I believe you can get around this with [the `canHandleIntent` method](https://www.sanity.io/docs/structure-builder-reference#canHandleIntent-56e11465fc2c). Could you try adding that?

**Comment by ricokahler** [comment]
This should've been addressed by #2886. It's currently out in a canary. See [here](https://github.com/sanity-io/sanity/issues/2140#issuecomment-953048011) for more details.
